src/WebJobs.Script.WebHost/WebScriptHostManager.cs

```diff
@@ -324,5 +327,24 @@ public Task<bool> DelayUntilHostReady(int timeoutSeconds = ScriptConstants.HostT
 
             return hostReady;
         }
+
+        internal static async Task<bool> DelayUntilHostReady(WebHostResolver resolver)
+        {
+            if (DelayRequests)
+            {
+                // delay until we're ready to start processing requests
+                await Utility.DelayAsync(ScriptConstants.HostTimeoutSeconds, ScriptConstants.HostPollingIntervalMilliseconds, () =>
+                {
+                    return DelayRequests;
+                });
+            }
+
+            // need to ensure the host startup has been initiated
+            var manager = resolver.GetWebScriptHostManager();
+            var tIgnore = manager.EnsureHostStarted(CancellationToken.None);
```

Comment:
Still unclear to me as to why this needs to be called outside of the hosted service. Why wouldn't that call be sufficient?

Reply:
HostedService code is only run on WebApp startup, right? Specialization is triggered after startup, and the startup code is not run again.